x-pack/plugins/observability/public/components/shared/exploratory_view/configurations/default_configs.ts

```diff
@@ -5,48 +5,32 @@
  * 2.0.
  */
 
-import { AppDataType, ReportViewType } from '../types';
-import { getRumDistributionConfig } from './rum/data_distribution_config';
-import { getSyntheticsDistributionConfig } from './synthetics/data_distribution_config';
-import { getSyntheticsKPIConfig } from './synthetics/kpi_over_time_config';
-import { getKPITrendsLensConfig } from './rum/kpi_over_time_config';
+import { AppDataType, ReportViewType, SeriesConfig } from '../types';
 import { IndexPattern } from '../../../../../../../../src/plugins/data/common';
-import { getCoreWebVitalsConfig } from './rum/core_web_vitals_config';
-import { getMobileKPIConfig } from './mobile/kpi_over_time_config';
-import { getMobileKPIDistributionConfig } from './mobile/distribution_config';
-import { getMobileDeviceDistributionConfig } from './mobile/device_distribution_config';
-import { DataTypes, ReportTypes } from './constants';
+import { ReportConfigMap } from '../contexts/exploatory_view_config';
 
 interface Props {
   reportType: ReportViewType;
   indexPattern: IndexPattern;
   dataType: AppDataType;
+  reportConfigMap: ReportConfigMap;
 }
 
-export const getDefaultConfigs = ({ reportType, dataType, indexPattern }: Props) => {
-  switch (dataType) {
-    case DataTypes.UX:
-      if (reportType === ReportTypes.DISTRIBUTION) {
-        return getRumDistributionConfig({ indexPattern });
-      }
-      if (reportType === ReportTypes.CORE_WEB_VITAL) {
-        return getCoreWebVitalsConfig({ indexPattern });
-      }
-      return getKPITrendsLensConfig({ indexPattern });
-    case DataTypes.SYNTHETICS:
-      if (reportType === ReportTypes.DISTRIBUTION) {
-        return getSyntheticsDistributionConfig({ indexPattern });
-      }
-      return getSyntheticsKPIConfig({ indexPattern });
-    case DataTypes.MOBILE:
-      if (reportType === ReportTypes.DISTRIBUTION) {
-        return getMobileKPIDistributionConfig({ indexPattern });
-      }
-      if (reportType === ReportTypes.DEVICE_DISTRIBUTION) {
-        return getMobileDeviceDistributionConfig({ indexPattern });
-      }
-      return getMobileKPIConfig({ indexPattern });
-    default:
-      return getKPITrendsLensConfig({ indexPattern });
-  }
+export const getDefaultConfigs = ({
+  reportType,
+  dataType,
+  indexPattern,
+  reportConfigMap,
+}: Props): SeriesConfig => {
+  let configResult: SeriesConfig;
+
+  reportConfigMap[dataType].some((fn) => {
+    const config = fn({ indexPattern });
+    if (config.reportType === reportType) {
+      configResult = config;
+    }
+    return config.reportType === reportType;
+  });
```

Comment:
- Is it safe to assume `reportConfigMap[dataType]` will always be non nullable?
- Consider  `reportConfigMap[dataType].find` if it makes more sense than `some`.

Reply:
Yes, it will always be present